### PR TITLE
[FIX] stock: duplicating warehouse dependencies

### PR DESCRIPTION
**Pull request by nd-dew**
[FIX] stock: duplicating warehouse dependencies

Before this commit when duplicating a warehouse, its operation types (picking.type) wouldn't get copied. This commit ensures that new picking.types are created for the duplicate warehouse.

### [Reproduce]
- run odoo 17 with -i stock,mrp_subcontracting
- in Inventory/Configuration/Warehouses Duplicate a Warehouse
- Bug: in  Inventory/Configuration/OperationTypes picking types aren't duplicated 

opw-3674614

### CODE REVIEW

**Review by HANNICHE-Walid — APPROVED** [review]
LGTM (when runbot errors are fixed)
new picking types should be created for the new warehouse

File: addons/mrp_subcontracting/tests/test_subcontracting.py
Comment (nle-odoo) [inline addons/mrp_subcontracting/tests/test_subcontracting.py:41]: the gettattr require a security review, since the fields should exist (and you don't have a fallback anyway getattr would cause an error if the field is not defined) you probably can just use `record[route_type]` and not use attr (hopefully it's not flagged)

same things for the 2 getattr below

**Review by nle-odoo — APPROVED** [review]
looks good to me, if we don't want to copy these fields

File: addons/mrp_subcontracting/tests/test_subcontracting.py
Comment (nd-dew) [reply addons/mrp_subcontracting/tests/test_subcontracting.py:41]: interesting tip! thanks, lemme try

Comment (nd-dew) [reply addons/mrp_subcontracting/tests/test_subcontracting.py]: `wh_original[route_type]` doesn't require security review. Switching to it

**Comment by Whenrow** [comment]
We always wanted to rewrite this file. You gave use another good reason :D For a bugfix I don't see other good way to do this. Anyway, I would add ` copy=False` also on `*_rule_id` fields. For instance, if you copy a warehouse with your patch. The pull rule 
`WH/stock -> subcontracting location` is not duplicated

**Comment by nd-dew** [comment]
added `copy=False` to all M2O fields of `'stock.rule'`

**Comment by nd-dew** [comment]
`stock.move` ` rule_id` should not get `copy=False` right @Whenrow

**Comment by Whenrow** [comment]
robodoo r+
thank you

**Pull request closed without merging**
